### PR TITLE
chore: migrate GPv2Settlement.swap balances tests to Foundry

### PR DESCRIPTION
**Pull request by fedgiac**
## Description

See title.

JS allows you to be much more flexible in tests, so this code is a workaround that's only reasonable because there are only few balance values.

## Test Plan

CI.

## Related Issues

#119

### CODE REVIEW

File: test/GPv2Settlement/Swap/Balances.t.sol [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    function test_performs_a_swap_from_erc20_to_internal_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_INTERNAL);
+    }
+
+    function test_performs_a_swap_from_internal_to_internal_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_INTERNAL, GPv2Order.BALANCE_INTERNAL);
+    }
+
+    function performs_a_swap_with_the_specified_balances(bytes32 buyTokenBalance, bytes32 sellTokenBalance) private {
+        Vm.Wallet memory trader = vm.createWallet("trader");
```
Comment (mfw78) [inline test/GPv2Settlement/Swap/Balances.t.sol:39]: nit: already defined in `Helper.sol`

File: test/GPv2Settlement/Swap/Balances.t.sol [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_INTERNAL, GPv2Order.BALANCE_EXTERNAL);
+    }
+
+    function test_performs_a_swap_from_erc20_to_internal_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_INTERNAL);
+    }
+
+    function test_performs_a_swap_from_internal_to_internal_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_INTERNAL, GPv2Order.BALANCE_INTERNAL);
+    }
```
Comment (mfw78) [inline test/GPv2Settlement/Swap/Balances.t.sol:36]: nit: I do find it a little odd to have the `buyTokenBalance` defined first, as opposed to `sellTokenBalance` (given the ordering seen in most places for the `Order` struct, the logical path when conisidering token flow, and the definition in the original tests being `sellTokenBalance` first).

Comment (mfw78) [reply test/GPv2Settlement/Swap/Balances.t.sol:36]: And on second look, the test labelling is reversed when considering from the perspective of token flow.

**Review by mfw78 — COMMENTED** [review]
Looks good, though some nits on labelling / ordering

File: test/GPv2Settlement/Swap/Balances.t.sol (hunk 2 above)
Comment (fedgiac) [reply test/GPv2Settlement/Swap/Balances.t.sol]: I inverted the two.
Indeed in hindsight I'd invert the standard ordering to be buy then sell. Still, I've noticed when writing this test that I actually mixed up the two in the function input and only noticed and fixed it when running the test. So being consistent is better here.

**Review by mfw78 — APPROVED** [review]

**Pull request merged**